src/lib.rs

```diff
@@ -135,3 +135,21 @@ pub mod messaging;
 pub mod queueing_honey_badger;
 pub mod sync_key_gen;
 pub mod transaction_queue;
+
+/// Common supertraits.
+pub mod traits {
+    use std::fmt::Debug;
+    use std::hash::Hash;
+
+    /// A transaction, user message, etc.
```

Comment:
I would like to see a longer comment here, documenting why the trait bounds were chosen as they are, since this is such an integral piece.

Reply:
Hmm, I usually don't document why I chose the specific bounds of supertraits. I hear you 100% about more thorough documentation but I'm not sure what I would say other than "We needed all this stuff at some point, somewhere".

I'm at a loss. What do you think I should write?

Reply:
I assume there was a specific reason for each of these to be added at some point? I will admit that the current set seems fairly innocuous though.